tests/ui/perfherder/dropdown_menu_items_test.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  fireEvent,
+  waitForElement,
+  getByRole,
+  shallow,
+} from '@testing-library/react';
+import { DropdownMenu } from 'reactstrap';
+
+import { createDropdowns } from '../../../ui/perfherder/FilterControls';
+import DropdownMenuItems from '../../../ui/shared/DropdownMenuItems';
+
+// const dropdownOptions = [
+//   {
+//     selectedItem: 'try',
+//     updateData: () => {},
+//     pinned: ['try', 'autoland', 'bogus'],
+//     options: ['mozilla-inbound', 'autoland', 'mozilla-central', 'try'],
+//     namespace: 'Projects',
+//   },
+// ];
+
+const selectedItem = 'try';
+const pinned = ['try', 'autoland', 'bogus'];
+const options = ['mozilla-inbound', 'autoland', 'mozilla-central', 'try'];
+const namespace = 'Projects';
+
+const dropdownMenuItems = () =>
+  render(
+    <DropdownMenuItems
+      selectedItem={selectedItem}
+      updateData={() => {}}
+      options={options}
+      pinned={pinned}
+      namespace={namespace}
+    />,
+  );
+
+test('pinned items are shown first', async () => {
+  const { getByRole, getByText } = dropdownMenuItems();
+  // render(createDropdowns(dropdownOptions));
+
+  const component = render(
+    <DropdownMenuItems
+      selectedItem={selectedItem}
+      updateData={() => {}}
+      options={options}
+      pinned={pinned}
+      namespace={namespace}
+    />,
+  );
+
+  const button = await waitForElement(() => screen.find('dropdown'));
+  expect(button).toBeInTheDocument();
+  fireEvent.click(button);
+
+  // const dropdownItems = await waitForElement(() => getAllByRole('menuitem'));
```

Comment:
What happened when you got here? This looks good so far. To make this easier you could also add `data-testid` to the `li`. 

Instead of `dropdown` for the button on line 55, I think you could also `getByText(selectedItem)`.

Reply:
ah, that's a good point. I kept getting weird errors when trying to fire a click event, tried with different types of selectors as well.

`fireEvent.click(getByText(selectedItem));` returned:
```
    Found multiple elements with the text: try

    (If this is intentional, then use the `*AllBy*` variant of the query (like `queryAllByText`, `getAllByText`, or `findAllByText`)).
```

`fireEvent.click(getAllByText(selectedItem)[0]);` fails with:
```
    TypeError: this.props.context.toggle is not a function

      61 |   component.debug(getAllByText(selectedItem)[0]);
      62 | 
    > 63 |   fireEvent.click(getAllByText(selectedItem)[0]);
```

there was actually another error I was getting but I'm not sure how I encountered that, haven't been able to reproduce. I've been fiddling with this a lot

Reply:
I thought the second error above might be due to the mock `onClick` action: `toggle: () => {}`, but I think it may be something else

Reply:
> ah, that's a good point. I kept getting weird errors when trying to fire a click event, tried with different types of selectors as well.
> 
> `fireEvent.click(getByText(selectedItem));` returned:
> 
> ```
>     Found multiple elements with the text: try
> 
>     (If this is intentional, then use the `*AllBy*` variant of the query (like `queryAllByText`, `getAllByText`, or `findAllByText`)).

Ah, I bet that is because `try` is the text in the dropdown menu button *and* its in the dropdown list. It is looking for anything in the DOM (of that component) that contains that text so you'll need to use something else. Run down the list of query options and if nothing else works, use `data-testid` for the dropdown button.

Reply:
This is also where looking at Perfherder tests others have written can be helpful. Once you start to become more familiar with Perfherder code, it'll give you insight into how to design the tests and show you things you didn't know about the library. The `graphs_view_test` has tests for the `TestDataModal`.

Reply:
> I thought the second error above might be due to the mock onClick action: toggle: () => {}, but I think it may be something else 

Where is that toggle action mocked? Are you referring to the `updateData` prop?

> TypeError: this.props.context.toggle is not a function

With the `getByAll` do you know if the first item in that array of nodes is the button or the list item? Because when you select a list item, that calls `updateData` and toggles the list closed.

I'd try using another query first and if you still hit that error, check in the spectrum chat channel for the react testing library (its mentioned in the library docs) if googling it didn't net anything.